elastic/kibana
# [ResponseOps][Alerts] Merge useAlertsDataView implementations

## Summary

Removes TriggersActionsUI's and Cases's implementations of the `useAlertDataView(s)` hooks, refactors the main implementation (temporarily placed in `@kbn/alerts-ui-shared`) to remove `useEffects` and make the binary (siem/non siem) logic clearer and better documented.

## To verify

- Create rules that fire alerts in all solutions
- Check that the alerts tables in all solutions work (visit the Stack management > Alerts, Observability > Alerts, Security > Alerts pages)
- Check that the table columns and fields browser (`Fields` button at the left of the table toolbar) show fields coherent with the current view and solution

### References

Closes #189467

### Checklist

Delete any items that are not applicable to this PR.
